ec/src/msm/variable_base/mod.rs

```diff
@@ -127,4 +128,43 @@ impl VariableBase {
                     total
                 })
     }
+    /// Steaming multi-scalar multiplication algorithm with hard-coded chunk
+    /// size.
+    pub fn msm_chunks<G, F, I: ?Sized, J>(bases_stream: &J, scalars_stream: &I) -> G::Projective
+    where
+        G: AffineCurve<ScalarField = F>,
+        I: Iterable,
+        F: PrimeField,
+        I::Item: Borrow<F>,
+        J: Iterable,
+        J::Item: Borrow<G>,
+    {
+        assert!(scalars_stream.len() <= bases_stream.len());
+
+        // remove offset
+        let bases_init = bases_stream.iter();
+        let mut scalars = scalars_stream.iter();
+
+        // align the streams
+        // NOTE: Change skip() to iter_advance_by() when it becomes stable.
```

Comment:
```suggestion
        // TODO: change `skip` to `advance_by` once rust-lang/rust#7774 is fixed. 
        // See <https://github.com/rust-lang/rust/issues/77404>
```

Reply:
Actually, I feel it is not a bug (``fixed''), it is just a feature that is pending to be stable.